mobile/ios/ios-core.ts

```diff
@@ -652,16 +648,45 @@ class WinSocket implements Mobile.IiOSDeviceSocket {
 		this.close();
 	}
 
-	public receiveMessage(): IFuture<void> {
-		return(() => {
-			var reply = this.receiveMessageHelper();
-			if(!this.$mobileDevice.isDataReceivingCompleted(reply)) {
-				reply = this.receiveMessageHelper();
+	public receiveMessage(): IFuture<Mobile.IiOSSocketResponseData> {
```

Comment:
WinSocket and PosixSocket classes implament the same interface. The `receiveMessage` functon in PosixSocket must return feature so we need to return feature in WinSocket too.